File: tasks/java/medium/Calculator.java
// Java - Medium

package tasks.java.medium;

public class Calculator {
    // TODO: Implement the Calculator class
    public double add(int a, int b){
        return (double)a+b;
    }
    public double subtract(int a, int b){
        return (double)a-b;
    }
    public double multiply(int a, int b){
        return (double)a*b;
    }
    public double divide(int a, int b){
        if(b==0) {
            throw new IllegalArgumentException("Cannot divide by zero");
        }
        return (double) a/b;
    }

    public static void main(String[] args) {
        //! Don't change the tests in the main method!
        Calculator calc = new Calculator();

        System.out.println("Addition: " + calc.add(5, 3));
        System.out.println("Subtraction: " + calc.subtract(5, 3));
        System.out.println("Multiplication: " + calc.multiply(5, 3));
        System.out.println("Division: " + calc.divide(5, 3));
    }
}

File: tasks/typescript/medium/calculator.js
// TypeScript - Medium
var Calculator;
(function (Calculator) {
    function add(a, b) {
        return a + b;
    }
    Calculator.add = add;
    function subtract(a, b) {
        return a - b;
    }
    Calculator.subtract = subtract;
    function multiply(a, b) {
        return a * b;
    }
    Calculator.multiply = multiply;
    function divide(a, b) {
        if (b === 0) {
            console.warn("Warning: Division by zero is not allowed!");
            return null;
        }
        return a / b;
    }
    Calculator.divide = divide;
})(Calculator || (Calculator = {}));
// Test cases
console.log(Calculator.add(5, 3)); // Expected: 8
console.log(Calculator.subtract(5, 3)); // Expected: 2
console.log(Calculator.multiply(5, 3)); // Expected: 15
console.log(Calculator.divide(5, 3)); // Expected: 1.666...
console.log(Calculator.divide(5, 0)); // Expected: Warning message & null
